### ISSUE TITLE
Twitter embed block should expose options for how the tweet is displayed

### ISSUE THREAD

**Issue by aaronjorbin** [WordPress/gutenberg#2744]
The embed shortcode allows for controlling things like if media should be displayed, if the thread should be displayed, and the language of the follow buttons. These should be exposed as options for the tweet embed

**Comment by aduth** [comment]
>The embed shortcode allows for controlling things like if media should be displayed, if the thread should be displayed, and the language of the follow buttons.

Are these available in the default installation? Or are you referring to the [`[tweet]` shortcode included in Jetpack](https://jetpack.com/support/shortcode-embeds/)? The challenge with this is that the arguments controlling these are added server-side to the provider URL, not the URL of the embed (which itself is a provider argument):

https://github.com/Automattic/jetpack/blob/94dfd5b0a48ed2cc1332e4b09b615299bd674ade/modules/shortcodes/tweet.php#L97-L106

Example: `https://publish.twitter.com/oembed?maxwidth=525&maxheight=788&hide_media=true&url=https%3A%2F%2Ftwitter.com%2Fwpcincinnati%2Fstatus%2F910313465033691142`

I think the most reasonable option here would be to allow the embed block to store additional provider arguments (in its comment attributes) which are then applied during the `oembed_fetch_url` filter during its server-side rendering pass. These would also need to be included in the call to the [oEmbed proxy endpoint](https://core.trac.wordpress.org/ticket/40450) in previewing the embed in the editor.

**Comment by bretterer** [comment]
I think it would be nice to have options for the block that allow you to change any of these options.

https://dev.twitter.com/web/embedded-timelines/parameters

@aduth

**Comment by aduth** [comment]
Related progress here: #3032

**Comment by paaljoachim** [comment]
@ntsekouras Nik. Perhaps you have some view points on this issue?

**Comment by ivan-ottinger** [comment]
Apart from the possibility of having a setting for number of tweets to show - when a Twitter timeline is embedded (as described in https://github.com/WordPress/gutenberg/issues/19417), It would be great to have other settings available as well, e.g.: theme, language, width, height (settings available through https://publish.twitter.com/#) when embedding manually:

![Markup on 2021-11-25 at 15:12:55](https://user-images.githubusercontent.com/25105483/143456881-a7629c12-8a2e-462f-aecd-7cbd7c38a0dd.png)

**Comment by ricjcs** [comment]
> Apart from the possibility of having a setting for number of tweets to show - when a Twitter timeline is embedded (as described in #19417), It would be great to have other settings available as well, e.g.: theme, language, width, height (settings available through https://publish.twitter.com/#) when embedding manually:

I would also like to have these options, especially the height, as it was in the old twitter timeline.

**Comment by ellatrix** [comment]
Worth noting that the style option seems to be gone now. But yes, there's still a language option. Perhaps we should set it to the site language by default and allow it to be changed. Seems relatively easy to implement.

There's also an option now to embed the video of a tweet.

**Comment by karmatosed** [comment]
I would love to have a list of what options there are now and some feedback on if this is still something to consider then we can move this onward.

**Comment by jasmussen** [comment]
Coming back to this one, it's genuinely unclear to me what options are available for Twitter embeds in today's landscape. If there are any, do we trust them surviving between releases? If there are positive answers to both, then I would recommend this PR actually move right into development: there should be enough componentry to add toggles for video embeds, or dropdowns for language, to the inspector under "Settings". If a PR were to materialise, please add design labels if design feedback is helpful—I'd be happy to personally help review and provide any design asisstance in-review. But largely, if there is a path forward on this one, it is blocked on development.
